base/process.jl

```diff
@@ -72,6 +72,13 @@ backticks, e.g.
 """
 Cmd
 
+hash(x::Cmd) = hash(x.exec)
```

Comment:
Hashing should be consistent with equality, so should probably be a function of all fields (including `ignorestatus` I think? which you're not considering below in `==` at the moment)